filebeat/input/filestream/filestream.go

```diff
@@ -89,13 +92,14 @@ func newFileReader(
 		closeInactive:      closerConfig.OnStateChange.Inactive,
 		closeRemoved:       closerConfig.OnStateChange.Removed,
 		closeRenamed:       closerConfig.OnStateChange.Renamed,
-		offset:             offset,
-		lastTimeRead:       time.Now(),
 		backoff:            backoff.NewExpBackoff(canceler.Done(), config.Backoff.Init, config.Backoff.Max),
 		readerCtx:          readerCtx,
 		tg:                 tg,
 	}
 
+	l.offset.Store(offset)
+	l.lastTimeRead.Store(time.Now().UnixNano())
```

Comment:
How do things behave if:

1. we strore the offset
2. go routine is put to sleep
3. Some other OS thread updates the file, maybe appends.
4. go routine wakes back up and records lastTimeRead as "Now"